pkg/api/types.go

```diff
@@ -822,6 +822,10 @@ type PodSpec struct {
 	// used must be specified.
 	// Optional: Default to false.
 	HostNetwork bool `json:"hostNetwork,omitempty"`
+	// PullSecretNames is an optional list of secret names to use for pulling any of the images used by this PodSpec.
+	// If specified, these secrets will be passed to individual puller implementations for them to use.  For example,
+	// in the case of docker, only DockerConfig type secrets are honored.
+	PullSecretNames []string `json:"pullSecretNames,omitempty" description:"list of secret names available for pulling the container images"`
```

Comment:
----- Original Message -----
> > @@ -822,6 +822,10 @@ type PodSpec struct {
> >  	// used must be specified.
> >  	// Optional: Default to false.
> >  	HostNetwork bool `json:"hostNetwork,omitempty"`
> > +	// PullSecretNames is an optional list of secret names to use for pulling
> > any of the images used by this PodSpec.
> > +	// If specified, these secrets will be passed to individual puller
> > implementations for them to use.  For example,
> > +	// in the case of docker, only DockerConfig type secrets are honored.
> > +	PullSecretNames []string `json:"pullSecretNames,omitempty"
> > description:"list of secret names available for pulling the container
> > images"`
> 
> What is the planned behavior for the admission controller if the pod is
> POSTed with some values in this array?  Is it to only use the user-provided
> values, or to merge?  I think the preferred behavior would be to not set any
> values if the user has set any.
> 
> In that case, this would ideally be a pointer to a slice rather than a slice.
> That way the admission controller can detect the difference between unset
> and explicitly empty list.

JSON and our conversion framework already preserves nil slice vs empty slice (or did).  So we don't have to have pointer to slice.